Library/Homebrew/extend/ENV/super.rb

```diff
@@ -28,8 +28,7 @@ def self.extended(base)
   def self.bin
     return unless MacOS.has_apple_developer_tools?
 
-    bin = HOMEBREW_ENV_PATH.subdirs.reject { |d| d.basename.to_s > MacOS::Xcode.version }.max
-    bin.realpath unless bin.nil?
+    HOMEBREW_ENV_PATH.subdirs.max.realpath
```

Comment:
> A bit unfair considering this is the same logic as before without the `.reject`.

Not unfair at all! It actually discloses a bug in the refactoring (and my `HOMEBREW_ENV_PATH/"scm"` suggestion was also bogus). With the `.reject` and the comparison:

- It's clearer what is being picked here and what is the criterion.
- It actually rejects all non-version directories like `pkgconfig` and `scm`.

Hence the correct refactoring of this for a generic OS would be to always return `nil`.